<!-- -*- mode: XML -*- -->
<launch>
	<!-- nodes

	<node pkg="rad_grab" type="radfront_delphi" name="radfront_delphi" output="screen" cwd="node" respawn="true" />
	
 -->

	<node pkg="rad_grab" type="radfrontsub_can" name="radfrontsub_can" output="screen" cwd="node" respawn="true" />
	<node pkg="rad_grab" type="radfrontsub_bbox" name="radfrontsub_bbox" output="screen" cwd="node" respawn="true" />
	<node pkg="rad_grab" type="radfrontsub_pcloud" name="radfrontsub_pcloud" output="screen" cwd="node" respawn="true" />

	<!-- Global arg ************************************************************ -->
	<arg name="radar_enable" default="1" />
	<arg name="delphi_enable" default="1" />
	<arg name="umec_enable" default="0" />
	<arg name="alpha_front_enable" default="0" />
	<arg name="alpha_side_enable" default="0" />

	<rosparam param="debug_output">true</rosparam>

	<!-- ******** Ros Parameter For Radar Stitching ******** -->
	<rosparam param="Delphi_Front_Center_Param">[0, 0, 0, 0, 0, 0]</rosparam>
	<rosparam param="Umec_Front_Center_Param">[0, 0, 0, 0, 0, 0]</rosparam>
	<rosparam param="Alpha_Front_Center_Param">[1, 0, 0, 0, 0, 0]</rosparam>
	<rosparam param="Alpha_Front_Right_Param">[1, 0, 0, 0, 0, 0]</rosparam>
	<rosparam param="Alpha_Front_Left_Param">[1, 0, 0, 0, 0, 0]</rosparam>
	<rosparam param="Alpha_Side_Right_Param">[1, 0, 0, 0, 0, 0]</rosparam>
	<rosparam param="Alpha_Side_Left_Param">[1, 0, 0, 0, 0, 0]</rosparam>
	<rosparam param="Alpha_Back_Right_Param">[1, 0, 0, 0, 0, 0]</rosparam>
	<rosparam param="Alpha_Back_Left_Param">[1, 0, 0, 0, 0, 0]</rosparam>

	<!-- ************ Radar Detection ************ -->
	<group ns="RadarDetection" if="$(eval arg('radar_enable') != 0)">

		<node pkg="rad_grab" type="radar_all" name="radar_all" output="screen" cwd="node" respawn="true" />

	</group>

	<!-- ************ Delphi Front Center ************ -->
	<group ns="DelphiFrontCenter" if="$(eval arg('delphi_enable') != 0)">

		<node pkg="rad_grab" type="radfront_delphi" name="radfront_delphi" output="screen" cwd="node"/>

	</group>

	<!-- ************ Alpha Front Center ************ -->
	<group ns="AlphaFrontCenter" if="$(eval arg('delphi_enable') != 0)">
		<arg name="filter_id" default="1" />
		<node pkg="rad_grab" type="radfront_alpha" name="radfront_alpha" output="screen" cwd="node">
			<param name="~/filter_id" value="$(arg filter_id)" />
		</node>
	</group>

	<!-- ************ Alpha Front Left ************ -->
	<group ns="AlphaFrontLeft" if="$(eval arg('alpha_front_enable') != 0)">
		<arg name="filter_id" default="2" />
		<node pkg="rad_grab" type="radfront_alpha" name="radfront_alpha" output="screen" cwd="node">
			<param name="~/filter_id" value="$(arg filter_id)" />
		</node>
	</group>

	<!-- ************ Alpha Front Right ************ -->
	<group ns="AlphaFrontRight" if="$(eval arg('alpha_front_enable') != 0)">
		<arg name="filter_id" default="3" />
		<node pkg="rad_grab" type="radfront_alpha" name="radfront_alpha" output="screen" cwd="node" respawn="true">
			<param name="~/filter_id" value="$(arg filter_id)" />
		</node>
	</group>

	<!-- ************ Alpha Side Left ************ -->
	<group ns="AlphaSideLeft" if="$(eval arg('alpha_side_enable') != 0)">
		<arg name="filter_id" default="4" />
		<node pkg="rad_grab" type="radfront_alpha" name="radfront_alpha" output="screen" cwd="node" respawn="true">
			<param name="~/filter_id" value="$(arg filter_id)" />
		</node>
	</group>

	<!-- ************ Alpha Side Right ************ -->
	<group ns="AlphaSideRight" if="$(eval arg('alpha_side_enable') != 0)">
		<arg name="filter_id" default="5" />
		<node pkg="rad_grab" type="radfront_alpha" name="radfront_alpha" output="screen" cwd="node" respawn="true">
			<param name="~/filter_id" value="$(arg filter_id)" />
		</node>
	</group>

	<!-- ************ Alpha Back Left ************ -->
	<group ns="AlphaBackLeft" if="$(eval arg('alpha_side_enable') != 0)">
		<arg name="filter_id" default="6" />
		<node pkg="rad_grab" type="radfront_alpha" name="radfront_alpha" output="screen" cwd="node" respawn="true">
			<param name="~/filter_id" value="$(arg filter_id)" />
		</node>
	</group>

	<!-- ************ Alpha Back Right ************ -->
	<group ns="AlphaBackRight" if="$(eval arg('alpha_side_enable') != 0)">
		<arg name="filter_id" default="7" />
		<node pkg="rad_grab" type="radfront_alpha" name="radfront_alpha" output="screen" cwd="node" respawn="true">
			<param name="~/filter_id" value="$(arg filter_id)" />
		</node>
	</group>


</launch>
